spyder/app/mainwindow.py

```diff
@@ -673,48 +681,59 @@ def create_edit_action(text, tr_text, icon):
         self.file_menu = self.menuBar().addMenu(_("&File"))
         self.file_toolbar = self.create_toolbar(_("File toolbar"),
                                                 "file_toolbar")
+        self.file_menu.aboutToShow.connect(self.verifyMyMenu)

```

Comment:
@juanis2112 please change this following the new signature of `verifyMyMenu` -> `verify_menu_actions`. To pass the `menu_actions` parameter you can use a lambda expression, something like:

```python
self.file_menu.aboutToShow.connect(lambda menu_actions=self.file_menu_actions: self.verify_menu_actions(menu_actions))
```

Reply:
I was about to make this suggestion too, well thought @dalthviz!

@juanis2112, I think you should also move all these statements of connecting to `aboutToShow/aboutToHide` to a new method called `setup_menus` and move all `aboutToShow//aboutToHide` calls there (including the ones you added, of course).